### PR TITLE
improve test run output (print panics)

### PR DESCRIPTION
**Pull request by PiotrTrzpil**
Now the output will also print panics. Example:

```
Failed tests and panics: ---------------------

19782:panic: 
19783-assert: mock: I don't know what to return because the method call was unexpected.
19784-  Either do Mock.On("HeadByNumber").Return(...) first, or remove the HeadByNumber() call.
--
130983:panic: 
130984-assert: mock: I don't know what to return because the method call was unexpected.
130985- Either do Mock.On("SubscribeFilterLogs").Return(...) first, or remove the SubscribeFilterLogs() call.

19810:FAIL      github.com/smartcontractkit/chainlink/core/internal     12.358s
131032:FAIL     github.com/smartcontractkit/chainlink/core/web  3.646s

----------------------------------------------

(Note: panics abruptly end the test run and may appear out-of-place. 
Finding their origin test may involve disabling parallelism)

Potentially related:

2021-07-07T15:14:22Z [ERROR] Operation failed                                   postgres/transaction.go:78 elapsed=0.458903 err=driver: bad connection rows=0 sql=SET LOCAL lock_timeout = 15000; SET LOCAL idle_in_transaction_session_timeout = 3600000; stacktrace=github.com/smartcontractkit/chainlink/core/services/postgres.GormTransactionWithDefaultContext.func1

```

### CODE REVIEW

File: core/services/log/models.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ func (b *broadcast) SetDecodedLog(newLog interface{}) {
 	b.decodedLog = newLog
 }
 
-func (b broadcast) JobID() JobIdSelect {
+func (b *broadcast) JobID() JobIdSelect {
```
Comment (PiotrTrzpil) [inline core/services/log/models.go:56]: unrelated to the PR - other methods already used pointer receiver

File: core/services/run_manager_test.go [hunk 2]
```diff
@@ -339,6 +339,10 @@ func TestRunManager_Create_fromRunLog_Happy(t *testing.T) {
 			ethClient.On("SubscribeNewHead", mock.Anything, mock.Anything).Return(sub, nil)
 			sub.On("Err").Return(nil)
 			sub.On("Unsubscribe").Return()
+			b := types.NewBlockWithHeader(&types.Header{
+				Number: big.NewInt(1),
+			})
+			ethClient.On("BlockByNumber", mock.Anything, mock.Anything).Maybe().Return(b, nil)
```
Comment (PiotrTrzpil) [inline core/services/run_manager_test.go:345]: fixes flakiness

File: tools/bin/go_core_tests [hunk 3]
```diff
@@ -4,8 +4,23 @@ set +e
...
+grep --line-number --after-context=2 "panic: " output.txt
+echo ""
+grep --line-number "\-\-\- FAIL" output.txt
+grep -n "FAIL	" output.txt
+echo ""
+echo "----------------------------------------------"
+
+echo ""
+echo "(Note: panics abruptly end the test run and may appear out-of-place, not where they originate."
+echo "Finding their origin test may involve disabling parallelism)"
```
Comment (samsondav) [inline tools/bin/go_core_tests:19]: This is my least fave thing about go's test suite. Absolutely horrible developer experience

**Review by samsondav — APPROVED** [review]

File: tools/bin/go_core_tests (hunk 3 above)
Comment (PiotrTrzpil) [reply tools/bin/go_core_tests:19]: yup, no isolation between tests unfortunately

**Pull request merged**